### PR TITLE
Automated TC 9225: Delete clones with krbd IO

### PR DESCRIPTION
**Pull request by VasishtaShastry**
Automated 9225: Delete clones with krbd IO

    1. Create a pool, an Image, 10 snapshots, 10 clones.
    2. generate IO on the Image, in parallel delete the clones

Changes:
1. Add test to suite suites/quincy/rbd/tier-1_rbd.yaml
2. Module handling the TC tests/rbd/delete_clones_with_io.py
3. Added a debug message tests/rbd/krbd_io_handler.py
4. Rectified errors in a method tests/rbd/rbd_utils.py

Signed-off-by: Vasishta <vashastr@redhat.com>

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by VasishtaShastry** [comment]
Log http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-ZC6NOS

File: tests/rbd/delete_clones_with_io.py
Comment (rkhambad) [inline tests/rbd/delete_clones_with_io.py:45]: we can use initial_rbd_config method here

Comment (VasishtaShastry) [reply tests/rbd/delete_clones_with_io.py:45]: Will implement once #2001 is merged.

**Comment by VasishtaShastry** [comment]
Log with EC Pool coverage - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-WCRB2Q

**Comment by VasishtaShastry** [comment]
Waiting for pull/2057 to merged to avoid merge conflict/dependency.

File: tests/rbd/delete_clones_with_io.py
Comment (manasagowri) [inline tests/rbd/delete_clones_with_io.py:53]: Why not use the initial_rbd_config method for pool and image creation?

**Review by manasagowri — CHANGES_REQUESTED** [review]
Just one small change. Rest looks good.

File: tests/rbd/delete_clones_with_io.py
Comment (VasishtaShastry) [reply tests/rbd/delete_clones_with_io.py:53]: Currently initial_rbd_config doesn't support `no config - all default arguments mode`
I've draft changes for them, will raise a PR in 2-3 days.

Can we please merge this PR (to keep tier-1 100% goal intact for Q4-22) ?

Comment (manasagowri) [reply tests/rbd/delete_clones_with_io.py:53]: Sure that should be okay.

**Review by manasagowri — APPROVED** [review]
LGTM

**Pull request merged**
